nxp_s32k/libuavcan/s32k_libuavcan.hpp

```diff
@@ -308,7 +308,7 @@ class S32K_InterfaceGroup : public InterfaceGroup<CAN::Frame<CAN::TypeFD::MaxFra
      * @param instance The FlexCAN peripheral instance number in which the ISR will be executed, starts at 0.
      *                 differing form this library's interface indexes that start at 1.
      */
-    static void __attribute__((interrupt("IRQ"))) S32K_libuavcan_ISR_handler(std::uint8_t instance)
+    static void S32K_libuavcan_ISR_handler(std::uint8_t instance)
```

Comment:
But you didn't add an attribute to the actual ISRs at the bottom of this file. Please refer to your compiler manual for details. For example, GCC: https://gcc.gnu.org/onlinedocs/gcc/ARM-Function-Attributes.html

Reply:
i apologize fo the late explanation, the reason for this __attribute__ was because the compiler with any optimizations turned on was discarding this function (it didn't appear in the .map file), with this, the function does appear in the map file but later realized the demo still didn't work with it.
I have version that works with -O3 next to be pushed. 

Regarding the actual ISR's, these must match the names declared in the startup code ["Link"](https://github.com/noxuz/libuavcan_demo/blob/master/Project_Settings/Startup_Code/startup_S32K146.S#L135) and these must be void-void functions that are handled by the NVIC hardware, and what goes in the body of these handlers its runned as usual code.

After, in link time, the address of the implemented function is correctly placed in the interrupt vector, so the __attribute isn't needed after all, due to the function being nested.
Although for ARMCC, the next link explains a bit further in the topic. [Link](http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dui0774g/vvi1465810473505.html)

Now, regarding the proper encapsulation of this function, Is is required to be public in order for the actual IRQ handler to be able to call it, but it cant be stand-alone because it uses attributes and other helper functions from within the class. Is this implementation correct or there shouldn't be additional public functions in the interface.

Reply:
The proper encapsulation is now fixed, view https://github.com/noxuz/platform_specific_components/commit/2622c26e715d8dec6e43cbb16149e240a77df8b6